### PR TITLE
Initial support for multiple elections in ramp refilings

### PR DESCRIPTION
**Pull request by kahlouie**
Connects #5776

### CODE REVIEW

File: app/models/ramp_refiling.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def create_end_product_and_contentions!
   end
 
   def election_receipt_date
-    ramp_election && ramp_election.receipt_date
+    ramp_elections && ramp_elections.map(&:receipt_date).min
```
[the remark this answers is not in the document]
Comment (kahlouie) [reply app/models/ramp_refiling.rb]: @shanear I should check that it's not empty though. Good catch.

[the remark this answers is not in the document]
Comment (shanear) [reply app/models/ramp_refiling.rb]: you don't have to do that, this method will return `nil` in that case:

```
irb(main):002:0> [].min
=> nil
```

which we can check for like we used to in the validation

File: app/models/ramp_refiling_intake.rb [hunk 2]
```diff
@@ -44,12 +46,16 @@ def review_errors
   end
 
   def ui_hash
+    issues = []
+    ramp_elections.each do |election|
+      issues += election.issues
+    end
```
[the remark this answers is not in the document]
Comment (kahlouie) [reply app/models/ramp_refiling_intake.rb]: Just moved it inline to where it was being used!

File: app/models/ramp_refiling.rb [hunk 3]
```diff
@@ -45,6 +45,10 @@ def needs_end_product?
 
   private
 
+  def ramp_elections
+    RampElection.established.where(veteran_file_number: veteran_file_number).all
```
[the remark this answers is not in the document]
Comment (kahlouie) [reply app/models/ramp_refiling.rb]: After discussion with @shanear in person, we decided that this is the correct call as we don't know which ramp elections will be connected to the ramp refiling yet.

**Pull request merged**